pyiron_atomistics/atomistics/structure/has_structure.py

```diff
@@ -83,6 +87,12 @@ def get_structure(self, frame=-1, wrap_atoms=True, iteration_step=None):
         """
         if iteration_step is not None:
             frame = iteration_step
+        if not isinstance(frame, int):
```

Comment:
I always run into headaches comparing for `int`yness. Can you make an array of numpy ints and check if this works as expected with those? If not, did numpy ints work OK before? In the case that traditional and numpy ints either both work now or that numpy ints never worked then I don't mind this, but we should avoid breaking numpy int interpretation.

Reply:
@liamhuber I found [this gem](https://docs.python.org/3/library/numbers.html#numbers.Integral) in stdlib and checked that it works with numpy ints, so I'll use it.

@samwaseda Since it's an abstract mixin testing it directly is a bit tricky.  I could either add a dummy implementation or test the implementors in the current code base.  Thoughts?  I think it's orthogonal to this issue though and would make it in a separate PR.

Reply:
my point is that if you're anyway checking the behaviour with both `int` and `np.int` (maybe also `np.int64`?), you can simply leave the lines in the unit test as well.

Reply:
And indeed:

```
i = np.int32(5)
a = np.arange(10)
if isinstance(i, int):
    print(a[i])
else:
    print(f'{i} is not an integer')
```

This prints `5 is not an integer`, even though `a[i]` very well returns 5.

Reply:
ok but we can come up with several scenarios, like `numbers` for some reason gets deprecated, or there's a major overhaul for this code or whatsoever, where someone thinks of rewriting these lines while believing they are maintaining the idea (i.e. `frame` is an `int`), even though in reality they might by mistake replace `numbers.Integral` by `int`, in which case the presence of a unit test containing `np.int32` etc. might be a reminder that `int` is not sufficient for this `if`-statement.

Reply:
I'm not 100% if I understood your comment, but if you're saying that for the testing you need an underlying structure, I guess the easiest solution is to overload `_get_structure` for the testing purpose.

Reply:
> I'm not 100% if I understood your comment, but if you're saying that for the testing you need an underlying structure, I guess the easiest solution is to overload `_get_structure` for the testing purpose.

Jep, that's what I meant.  It's possible to just sub classes for the test and I can do it, but as I said I'd rather leave it for another day.